pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/PersistentTopics.java

```diff
@@ -1458,6 +1458,33 @@ public PersistentOfflineTopicStats getBacklog(
         return internalGetBacklog(authoritative);
     }
 
+    @GET
+    @Path("/{tenant}/{namespace}/{topic}/messageId/{messageId}/checkLedgerExists/{checkLedgerExists}/backlogSize")
```

Comment:
This API is confusing. Why is a message ID required? If it's not provides can it default to the latest?

This `checkLedgerExists/{checkLedgerExists}` should not be part of the path in my opinion. If you need this `checkLedgerExists` it should be passed as a query parameter. 

I still don't understand why `checkLedgerExists` is needed. Can't the response return an error if it does not exist?

Reply:
For multi-partitions topic, We cannot return error if messageId not exists on one partitioned-topic, instead, We should calculate size of ledger larger than messageId's ledger id

Reply:
For partitioned-topic, We should return an error if ledger not exists on it. 
 But for  multi-partitions topic, We return an error when ledger not exists on all partitions. We should include  those partitions which ledger not exists into calculate